source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidRealTimeMultiplayerClient.cs

```diff
@@ -34,111 +56,689 @@ public AndroidRealTimeMultiplayerClient(AndroidJavaObject account)
             ulong exclusiveBitMask,
             RealTimeMultiplayerListener listener)
         {
-            AndroidJavaObject roomUpdateCallback = new AndroidJavaObject(
-                "com.google.games.bridge.RoomUpdateCallbackProxy",
-                new RoomUpdateCallbackProxy( /* parent= */this, listener));
-            AndroidJavaObject realTimeListener = null; //listener
-            // build room config
-            using (var roomConfigClass =
-                new AndroidJavaClass("com.google.android.gms.games.multiplayer.realtime.RoomConfig"))
+            lock (mSessionLock)
             {
+                if (GetRoomStatus() == RoomStatus.Active)
+                {
+                    OurUtils.Logger.e("Received attempt to create a new room without cleaning up the old one.");
```

Comment:
Should we call provided listener with some failed result?
listener.OnRoomConnected(false);

Reply:
Yes we should.